src/alfons/textShaper.cpp

```diff
@@ -310,7 +310,7 @@ bool TextShaper::processRun(const FontFace& _face, const TextRun& _run,
         hb_codepoint_t codepoint = glyphInfos[pos].codepoint;
         uint32_t clusterId = glyphInfos[pos].cluster;
 
-        auto id = clusterId;
+        uint32_t id;
```

Comment:
What did Xcode complain about here? Using `auto` is nice because it ensures that `id` has the same type as `clusterId`, even if that type changes later.

Reply:
It was not about `auto`, it was the value of id set to `clusterId` which was never used in the following code, but was rather updated in the next `if..else` block. It saves an unnecessary `store`  operation. (Which are some perf tricks I recently learned in my quest to learn more on data-driven programming)

Reply:
Its reported as a "Dead Store".

Reply:
Also for purposes of ensuring that `id` has  the same type as `cluster` we should use `decltype`. http://en.cppreference.com/w/cpp/language/decltype

Reply:
Ok - I suppose we could have it both ways with something like `decltype(clusterId) id;` but it's probably fine the way it is.